models/lfs.go

```diff
@@ -44,20 +44,20 @@ const (
 func NewLFSMetaObject(m *LFSMetaObject) (*LFSMetaObject, error) {
 	var err error
 
-	has, err := x.Get(m)
+	sess := x.NewSession()
+	defer sess.Close()
+	if err = sess.Begin(); err != nil {
+		return nil, err
+	}
+
+	has, err := sess.Get(m)
 	if err != nil {
 		return nil, err
 	}
 
 	if has {
 		m.Existing = true
-		return m, nil
-	}
-
-	sess := x.NewSession()
-	defer sess.Close()
-	if err = sess.Begin(); err != nil {
-		return nil, err
+		return m, sess.Commit()
```

Comment:
Maybe we want to update `m` before `sess.Commit`?

Reply:
Hmm... This branch never updated m before. The `sess.Commit()` is only there because I've moved the opening of the transaction to as soon as we check if an LFS meta object is in the db, not just when we're creating new lfs meta objects after not finding them in the db. If we need to insert I do update m.

Checking if an object is in the db and then inserting if it's not should be a in single transaction otherwise you can have race conditions.